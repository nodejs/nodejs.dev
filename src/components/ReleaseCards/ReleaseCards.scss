.release-card__row {
  display: flex;
  align-items: center;
  justify-content: space-between;
  list-style: none;
  padding: 0px;
  margin-block-end: 0px;
  margin-block-start: 0px;
  width: 100%;
  max-width: 615px;
  margin-top: 40px;
}

.release-card--active,
.release-card--inactive {
  background-color: var(--color-fill-app);
  border: 1px solid var(--color-border-secondary);
  display: flex;
  flex-direction: column;
  padding: 16px;
  width: 179px;
  height: 167px;
  box-sizing: border-box;
  box-shadow: 5px 10px 50px rgba(0, 0, 0, 0.05);
  border-radius: 4px;
  cursor: pointer;
}

.release-card--active {
  height: 197px;
  border-radius: 4px 4px 0px 0px;
  border-top: 4px solid var(--brand5);
}

.release-card__top {
  display: flex;
  justify-content: space-between;
}

.release-card__top--inactive {
  width: 48px;
  height: 48px;
}

.release-card__top--active {
  width: 56px;
  height: 56px;
}

.release-card__link {
  height: 24px;
  width: 24px;
  color: var(--black5);
  font-size: 24px;
}

.release-card__label--active {
  margin-top: 24px;
  font-weight: var(--font-weight-semibold);
}

.release-card__label--inactive {
  margin-top: 16px;
  font-weight: var(--font-weight-semibold);
}

.release-card__filename {
  font-weight: normal;
  font-size: 12px;
  line-height: 18px;
}
