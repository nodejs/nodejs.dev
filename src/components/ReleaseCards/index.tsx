import React from 'react';
import { ReleaseData } from '../../hooks/useReleaseHistory';

interface Props {
  line?: ReleaseData;
}

export default function ReleaseCards({ line }: Props): JSX.Element {
  const fileName = line && line.version;
  return (
    <div>
      <div>
        <i className="material-icons">cloud</i>
        <p>Windows Installer</p>
        <a
          href={`https://nodejs.org/dist/${fileName}/node-${
            line && line.version
          }-x86.msi`}
        >
          node-
          {line && line.version}
          .x86.msi
        </a>
      </div>
      <div>
        <i className="material-icons">cloud</i>
        <p>Mac Installer</p>
        <a
          href={`https://nodejs.org/dist/${fileName}/node-${
            line && line.version
          }.pkg`}
        >
          node-
          {line && line.version}
          .pkg
        </a>
      </div>
      <div>
        <i className="material-icons">cloud</i>
        <p>Source Code</p>
        <a
          href={`https://nodejs.org/dist/${fileName}/node-${
            line && line.version
          }.tar.gz`}
        >
          node-
          {line && line.version}
          .tar.gz
        </a>
      </div>
    </div>
  );
}
