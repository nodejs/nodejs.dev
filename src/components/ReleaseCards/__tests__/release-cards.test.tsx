import React from 'react';
import { render } from '@testing-library/react';
import ReleaseCards from '..';

describe('ReleaseCards component', () => {
  it('renders correctly with data', () => {
    const mockLineData = {
      version: 'v14.13.0',
      date: '2020-09-29',
      files: ['aix-ppc64', 'headers', 'linux-arm64', 'linux-armv7l'],
      npm: '6.14.8',
      v8: '8.4.371.19',
      uv: '1.40.0',
      zlib: '1.2.11',
      openssl: '1.1.1g',
      modules: '83',
      lts: false,
      security: false,
    };

    const { container } = render(<ReleaseCards line={mockLineData} />);
    expect(container).toMatchSnapshot();
  });
});
