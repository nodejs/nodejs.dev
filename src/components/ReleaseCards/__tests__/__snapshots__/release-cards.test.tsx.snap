// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseCards component renders correctly with data 1`] = `
<div>
  <div>
    <div>
      <i
        class="material-icons"
      >
        cloud
      </i>
      <p>
        Windows Installer
      </p>
      <a
        href="https://nodejs.org/dist/v14.13.0/node-v14.13.0-x86.msi"
      >
        node-
        v14.13.0
        .x86.msi
      </a>
    </div>
    <div>
      <i
        class="material-icons"
      >
        cloud
      </i>
      <p>
        Mac Installer
      </p>
      <a
        href="https://nodejs.org/dist/v14.13.0/node-v14.13.0.pkg"
      >
        node-
        v14.13.0
        .pkg
      </a>
    </div>
    <div>
      <i
        class="material-icons"
      >
        cloud
      </i>
      <p>
        Source Code
      </p>
      <a
        href="https://nodejs.org/dist/v14.13.0/node-v14.13.0.tar.gz"
      >
        node-
        v14.13.0
        .tar.gz
      </a>
    </div>
  </div>
</div>
`;
